datafusion/src/logical_plan/plan.rs

```diff
@@ -107,6 +107,19 @@ pub struct ExtensionPlan {
     pub node: Arc<dyn UserDefinedLogicalNode + Send + Sync>,
 }
 
+/// Aggregates its input based on a set of grouping and aggregate
+/// expressions (e.g. SUM).
+#[derive(Clone)]
+pub struct AggregatePlan {
```

Comment:
I think calling this `Aggregate` would be more consistent with what has been refactored in other PRs.

Let me know when this PR is ready @matthewmturner 

Thanks!